finagle-mysql/src/main/scala/com/twitter/finagle/mysql/ConnectionInitSql.scala

```diff
@@ -0,0 +1,35 @@
+package com.twitter.finagle.mysql
+
+import com.twitter.finagle._
+import com.twitter.finagle.mysql.param.ConnectionInitRequest
+import com.twitter.util._
+
+/**
+  * A stack module that executes a request when create a connection.
+  */
+object ConnectionInitSql {
+  val Role: Stack.Role = Stack.Role("ConnectionInitSql")
+
+  private[finagle] def module: Stackable[ServiceFactory[Request, Result]] =
+    new Stack.Module1[ConnectionInitRequest, ServiceFactory[Request, Result]] {
+      val role: Stack.Role = Role
+      val description: String = "Installs a connection init sql module in the stack"
+
+      def make(in: ConnectionInitRequest, next: ServiceFactory[Request, Result]): ServiceFactory[Request, Result] = {
+        in.request match {
+          case Some(req) =>
+            next.flatMap { service =>
```

Comment:
I would add a comment here that this assumes the module is added below the connection pool (otherwise, we would need to return it to the pool on the happy path too).

Reply:
add a comment https://github.com/twitter/finagle/pull/840/commits/d6532f8c6498bb8c871774180b586dfcaaa617b0